File: Java/1436_Destination_City.java
class Solution {
    public String destCity(List<List<String>> paths) {
        HashSet<String> toSet = new HashSet<String>();
        HashSet<String> fromSet = new HashSet<String>();
        for(List<String> path : paths){
            String from = path.get(0);
            String to = path.get(1);
            if(toSet.contains(from)){
                toSet.remove(from);
            }
            else{
                fromSet.add(from);
            }
            toSet.add(to);
        }
        for(String s : fromSet){
            if(toSet.contains(s)){
                toSet.remove(s);
            }
        }
        return toSet.iterator().next();
    }
}

File: CS/1436_Destination_City.cs
public class Solution {
    public string DestCity(IList<IList<string>> paths) {
        string result = "";
        HashSet<string> se = new HashSet<string>();
        for(int i = 0; i < paths.Count; i++){
            se.Add(paths[i][0]);
        }
        for(int i = 0; i < paths.Count; i++){
            if(!se.Contains(paths[i][1])){
            //if(se.Add(paths[i][1])){
                result = paths[i][1];
                break;
            }
        }
        return result;
    }
}
